services/apps/alcs/src/providers/typeorm/migrations/1692902429290-update_dec_number_generation_for_nois.ts

```diff
@@ -0,0 +1,54 @@
+import { MigrationInterface, QueryRunner } from 'typeorm';
+
+export class updateDecNumberGenerationForNois1692902429290
+  implements MigrationInterface
+{
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(
+      `CREATE OR REPLACE FUNCTION alcs.generate_next_resolution_number(p_resolution_year integer)
+          RETURNS integer
+          LANGUAGE plpgsql
+        AS $function$
+          
+            declare next_resolution_number integer;
```

Comment:
This function is almost perfectly symmetric.  I think it will be a good idea to add one or two sentences explaining what is happening here.